### ISSUE TITLE
Improvement: Ability to set SecurityContext

### ISSUE THREAD

**Issue by OS-marcelopereira** [projectcalico/calico#7282]
Improvement: Ability to set SecurityContext

## Expected Behavior
I have a usecase where I need to set a SecurityContext block on the APIServer deployment,  the apiserver container and componentResources, in order to comply with security requitements.

Looking at the documentation, I see that such a thing is not available for the [APIServerDeploymentContainer](https://projectcalico.docs.tigera.io/reference/installation/api#operator.tigera.io/v1.APIServerDeployment), [APIServerDeploymentContainer](https://projectcalico.docs.tigera.io/reference/installation/api#operator.tigera.io/v1.APIServerDeploymentContainer) and [APIServerDeploymentContainer](https://projectcalico.docs.tigera.io/reference/installation/api#operator.tigera.io/v1.ComponentResource)

Is this something that might be possible in the future?

Calico version: v3.24.5

**Comment by caseydavenport** [comment]
Yep, looks like we hard-code a security context here: https://github.com/tigera/operator/blob/master/pkg/render/apiserver.go#L940

It should be possible to expose an option for this, although we need to be careful not to expose an option that can easily break user's clusters.

**Comment by clayvan** [comment]
https://github.com/tigera/operator/issues/2720

**Comment by flo-mic** [comment]
is there any update on this?

**Comment by caseydavenport** [comment]
Just waiting on someone to pick up the torch for this one - I think it's generally agreed to be a good thing. Any takers?
